web/src/app/util/NumberField.tsx

```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from 'react'
+import { TextField, TextFieldProps } from '@material-ui/core'
+
+type NumberFieldProps = TextFieldProps & {
```

Comment:
I was in favor of `type` here instead of `interface` because `NumberField` should be a strict API for us to use within the app, so we want to use a `type` here to indicate while we're maintaining that we don't expect `NumberField` to ever be redeclared or extended.

[Here's a brief article I read that helped me make my decision](https://dev.to/reyronald/typescript-types-or-interfaces-for-react-component-props-1408), since in the past I've not really known when to use one vs. the other in React